spec/formatting.md

```diff
@@ -10,14 +10,18 @@ the successor to ICU MessageFormat, henceforth called ICU MessageFormat 1.0.
 
 ## Variable Resolution
 
-To resolve the value of a Variable,
-its Name is used to identify either a local variable,
+To resolve the value of a _variable_,
+its _name_ is used to identify either a local variable,
 or a variable defined elsewhere.
-If a local variable and an externally defined one use the same name,
-the local variable takes precedence.
 
-It is an error for a local variable definition to
-refer to a local variable that's defined after it in the message.
+If more than one _declaration_ binds a value to the same _name_,
+or if an externally defined variable and a _declaration_ use the same _name_,
+the resolved value of the most recent _declaration_ preceding the _variable_ is used.
+
+A _declaration_ MAY override the value of a _variable_ for later parts of the _message_.
```

Comment:
I think that last time we proposed to not allow overriding.
Compared to programming, these messages are a relatively short "code block"
There should be no problem to just use a different name.
It just introduces complexity without any benefit that I can imagine.